src/audio/asrc/asrc_ipc3.c

```diff
@@ -0,0 +1,1027 @@
+// SPDX-License-Identifier: BSD-3-Clause
```

Comment:
can we extract common code?

Reply:
is there any else to extract? Thanks for pointing out in detail.

Reply:
@lyakh , we can use another PR to do the extraction and ipc3/ipc4 isolation, this PR can only focus on module convertion.

Reply:
> @lyakh , we can use another PR to do the extraction and ipc3/ipc4 isolation, this PR can only focus on module convertion.

@btian1 sorry, I don't think it's a good approach to first duplicate the code to then merge it again